### PR TITLE
Allow specification of a custom interface name

### PR DESCRIPTION
**Pull request by queengooborg**
This PR adds a new `tsName` schema property that allows a developer to overwrite the default generated name.  This is super helpful in cases where two interfaces/types are generated with the same name (and one has a numerical increment), or where the developer may wish to ensure backwards compatibility of their TypeScript definitions.

Before:

```ts
/**
 * This interface was referenced by `Browsers`'s JSON-Schema
 * via the `definition` "browsers".
 */
export type Browsers1 = Record<string, BrowserStatement>;

export interface Browsers {
  browsers?: Browsers1;
}
```

After:

```ts
/**
 * This interface was referenced by `Browsers`'s JSON-Schema
 * via the `definition` "browsers".
 */
export type Browsers = Record<string, BrowserStatement>;

export interface BrowsersData {
  browsers?: Browsers;
}
```

This is related to #181 (and potentially resolves it?).  Unlike #215, however, this PR aims to grant the developer more finite control over the names.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -170,6 +170,7 @@ json2ts -i foo.json -o foo.d.ts --style.singleQuote --no-style.semi
 ## Custom schema properties:
 
 - `tsType`: Overrides the type that's generated from the schema. Useful for forcing a type to `any` or when using non-standard JSON schema extensions ([eg](https://github.com/sokra/json-schema-to-typescript/blob/f1f40307cf5efa328522bb1c9ae0b0d9e5f367aa/test/e2e/customType.ts)).
+- `tsName`: Overrides the name of a generated interface from the schema. Useful if two items are generated with similar names.
```
Comment (bcherny) [inline README.md:173]: Could you call this `x-name`? We'll be renaming `tsType` and `tsEnumNames` soon to match the new convention.

**Review by bcherny — CHANGES_REQUESTED** [review]
Seems reasonable! A few minor suggestions before we merge this.

File: test/e2e/customName.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      maxProperties: 1
+    }
+  },
+
+  type: 'object',
+  properties: {
+    grocery: {$ref: '#/definitions/grocery'}
+  },
+  additionalProperties: false,
+  tsName: 'GroceryList'
```
[the remark this answers is not in the document]
Comment (queengooborg) [reply test/e2e/customName.ts]: I didn't realize that `title` was an actual schema property!  In that case, the `title` property solves the problem for our use case, but I'll still update the tests and see this PR through to the end!

[the remark this answers is not in the document]
Comment (bcherny) [reply test/e2e/customName.ts]: If title works for your use case, we can also pause the work on this PR. No need to add more options unless there’s strong demand for them.

**Pull request closed without merging**